### ISSUE TITLE
Show which sources changed when running a task with verbose

### ISSUE THREAD

**Issue by abrahammurciano** [go-task/task#837]
Just like when you have a task with status checks, and you run it with -vv, it will log that it ran the command and show the status code, it would be useful if it also shows which sources were updated and which were not.

For example it would help me debug why a dependency of a task is running again after it was just run explicitly and all the status checks exited zero. I'd like to know which entry listed in sources it claims was changed so I can check if it was indeed changed and why

**Comment by ssbarnea** [comment]
I totally agree that this is highly needed. I already asked on https://github.com/go-task/task/discussions/2533 but got no answers yet.

We need a way to make taskfile report why it determined that things changed. I am aware that it does not store checksum of each file but at least it could store the list of files before and after so we could spot new files being added or removed that determine the final outcome.

**Comment by trulede** [comment]
I have a PR open (#2617) which introduces a "task" log, and that could be a good place to print this kind of detailed information. A single task level log including a list of changed files and associated metadata (checksum, timestamps etc). That kind of structured data could then be consumed by a script/tool (as well as task users ... perhaps).
